gabbi/case.py

```diff
@@ -544,7 +544,11 @@ def assert_in_or_print_output(self, expected, iterable):
                 dumper_class = self.get_content_handler(self.content_type)
                 if dumper_class:
                     full_response = dumper_class.dumps(self.response_data,
-                                                       pretty=True)
+                                                       pretty=True, test=self)
+                    # Sometimes, a given content handler may not hand back a
+                    # string for transmission, so we should be careful.
+                    if not isinstance(full_response, six.string_types):
+                        full_response = str(full_response)
```

Comment:
Was worried about this initially because in order to get my initial prototype to work for the multipart/form-data handler I had to pass the content along as a dict to the 'fields' param of urllib3's request. Since having context to the test means we can take advantage of properly stringifying the body and use the currently existing approach to call request (i.e. with just `body=` instead of `fields=`), I don't think even the other PR is necessary, at least not now. I'll take this out anyways.